sbi/utils/user_input_checks_utils.py

```diff
@@ -373,3 +373,69 @@ def build_support(
             support = constraints.interval(lower_bound, upper_bound)
 
     return support
+
+
+class OneDimPriorWrapper(Distribution):
```

Comment:
Looks great. I wonder if we can refactor the code to have one wrapper that converts a torch distribution to a distribution that is compatible with `sbi`. But I'm in favor of doing this in a new PR and pushing this one through for now.